catalog/spatial/registry/registry-report-viewer/src/test/java/org/codice/ddf/registry/report/viewer/RegistryReportViewerTest.java

```diff
@@ -0,0 +1,219 @@
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.registry.report.viewer;
+
+import static org.hamcrest.Matchers.hasItem;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Matchers.anyList;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Map;
+
+import javax.ws.rs.core.Response;
+import javax.xml.bind.JAXBElement;
+
+import org.codice.ddf.parser.Parser;
+import org.codice.ddf.parser.ParserConfigurator;
+import org.codice.ddf.parser.ParserException;
+import org.codice.ddf.parser.xml.XmlParser;
+import org.codice.ddf.registry.federationadmin.service.FederationAdminException;
+import org.codice.ddf.registry.federationadmin.service.FederationAdminService;
+import org.codice.ddf.registry.schemabindings.RegistryPackageUtils;
+import org.junit.Before;
+import org.junit.Test;
+
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.RegistryObjectType;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.RegistryPackageType;
+
+public class RegistryReportViewerTest {
+
+    private Parser parser;
+
+    private ParserConfigurator configurator;
+
+    private FederationAdminService federationAdminService = mock(FederationAdminService.class);
+
+    @Before
+    public void setUp() {
+        parser = new XmlParser();
+
+        configurator = parser.configureParser(Arrays.asList(RegistryObjectType.class.getPackage()
+                        .getName(),
+                RegistryPackageUtils.OGC_FACTORY.getClass()
+                        .getPackage()
+                        .getName(),
+                RegistryPackageUtils.GML_FACTORY.getClass()
+                        .getPackage()
+                        .getName()),
+                this.getClass()
+                        .getClassLoader());
+    }
+
+    @Test
+    public void testWithBlankMetacard() {
+        RegistryReportViewer reportViewer = new RegistryReportViewer();
+        Response response = reportViewer.viewRegInfoHtml(null, null);
+        assertThat(response.getStatusInfo()
+                .getStatusCode(), is(Response.Status.BAD_REQUEST.getStatusCode()));
+    }
+
+    @Test
+    public void testWithFederationAdminException() throws FederationAdminException {
+        when(federationAdminService.getRegistryObjectByMetacardId(anyString(),
+                anyList())).thenThrow(new FederationAdminException());
+        RegistryReportViewer reportViewer = new RegistryReportViewer();
```

Comment:
Looks like lines 80 and 81 could be extracted from each of the tests and placed in the setup method.